include/criterion/assert.h

```diff
@@ -30,109 +35,1041 @@
+ */
+
+/**
+ * Passes if Actual is byte-to-byte equal to Expected
+ *
+ * Call: \c cr_assert_arr_eq(Actual, Expected, [FormatString, [Args...]])\n
+ * Passes if Actual is byte-to-byte equal to Expected.
+ * Otherwise the test is marked as failure and the execution of the function
+ * is aborted.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual Array to test
+ * @param[in] Expected Expected array
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
 # define cr_assert_arr_eq(...) CR_EXPAND(cr_assert_mem_op_va_(CR_FAIL_ABORT_,      ==, __VA_ARGS__))
+
+/**
+ * Passes if Actual is byte-to-byte equal to Expected
+ *
+ * Call: \c cr_assert_arr_eq(Actual, Expected, [FormatString, [Args...]])\n
+ * Passes if Actual is byte-to-byte equal to Expected.
+ * Otherwise the test is marked as failure but the execution will continue.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual Array to test
+ * @param[in] Expected Expected array
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
+
 # define cr_expect_arr_eq(...) CR_EXPAND(cr_assert_mem_op_va_(CR_FAIL_CONTINUES_,  ==, __VA_ARGS__))
 
+
+/**
+ * Passes if Actual is not byte-to-byte equal to Expected
+ *
+ * Call: \c cr_assert_arr_neq(Actual, Expected, Size, [FormatString,
+ * [Args...]])\n
+ * Passes if Actual is not byte-to-byte equal to Expected.
+ * Otherwise the test is marked as failure and the execution of the function
+ * is aborted.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual Array to test
+ * @param[in] Expected Expected array
+ * @param[in] Size Number of bytes to check
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
 # define cr_assert_arr_neq(...) CR_EXPAND(cr_assert_mem_op_va_(CR_FAIL_ABORT_,     !=, __VA_ARGS__))
+
+/**
+ * Passes if Actual is not byte-to-byte equal to Expected
+ *
+ * Call: \c cr_assert_arr_neq(Actual, Expected, Size, [FormatString,
+ * [Args...]])\n
+ * Passes if Actual is not byte-to-byte equal to Expected.
+ * Otherwise the test is marked as failure but the execution will continue.
+ * The optional string is printed on failure.
+ *
+ * @param[in] Actual Array to test
+ * @param[in] Expected Expected array
+ * @param[in] Size Number of bytes to check
+ * @param[in] fmt printf like format string
+ *
+ *****************************************************************************/
+
 # define cr_expect_arr_neq(...) CR_EXPAND(cr_assert_mem_op_va_(CR_FAIL_CONTINUES_, !=, __VA_ARGS__))
 
-// Safe array comparison assertions
+/**@}*/
+
+/**
+ * @defgroup SafeArrCmpAsserts Safe array comparison assertions
+ * @{
+ */
 
 # if defined(__GNUC__) || defined(__clang__) || defined(__cplusplus)
 
+
+/**
+ * Passes if Actual is comparatively equal to Expected (C++ / GNU C99 only)
```

Comment:
The `(C++ / GNU C99 only)` should be a full sentence in a note below `The optional string is printed on failure.`, maybe something like:
```
@note
This macro is only available on C++ and GNU C compilers.
```